lib/improver/nowcasting/accumulation.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2019 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+This module defines the Accumulation class for calculating precipitation
+accumulations from advected radar fields. It also includes the
+AccumulationAggregator class for combining these into longer periods.
+"""
+import warnings
+import numpy as np
+
+import iris
+from improver.cube_combiner import CubeCombiner
+from improver.utilities.cube_units import (enforce_coordinate_units_and_dtypes,
+                                           enforce_diagnostic_units_and_dtypes)
+
+
+class Accumulation:
+    """
+    Class to calculate precipitation accumulations from radar rates fields
+    provided at discrete time intervals. The plugin will return one fewer
+    cubes of accumulation than the number of input precipitation rate cubes.
+    The time intervals between the precipitation rate cubes determine the
+    accumulation periods.
+    """
```

Comment:
Why is there a need for this intermediate stage?  Would it not be more logical for this plugin to create a complete accumulation from all the input files?